HW2/P4/driver.py

```diff
@@ -14,19 +14,54 @@
 import filtering
 from timer import Timer
 import threading
+from Queue import Queue
 
 def py_median_3x3(image, iterations=10, num_threads=1):
     ''' repeatedly filter with a 3x3 median '''
     tmpA = image.copy()
     tmpB = np.empty_like(tmpA)
-
+    Threads = []
+    Events = [] #np.empty([num_threads,iterations])
+    
+    # Create 2D array of events, each event corresponds to a single (thread, iteration) pair
+    for i in range(iterations):
+        for n in range(num_threads):
+            Events += [threading.Event()]
+    Events = np.array(Events).reshape((num_threads,iterations))
```

Comment:
You could also write this as
```events = [ [threading.Event() for n in range(num_threads)] for i in range(iterations) ]```,
it doesn't really matter but there probably isn't too much advantage of using np arrays with non-numeric types.